make it a fallback skill

JarbasAl opened the pull request:
fixes https://github.com/MycroftAI/mycroft-core/issues/1566

now padatious skills have a chance to use "stop words"

JarbasAl committed:
make it a fallback skill

forslund commented on __init__.py:
```diff
@@ -14,30 +14,51 @@
+        words = utterance.split(" ")
+        for stop_word in self.stop_words:
+            if stop_word in words:
+                self.emitter.emit(message.reply("mycroft.stop"))
```
The skill tests indicate that there's something afoot here. Seems the reply requires a data field

Just do a `self.emitter.emit(message.reply("mycroft.stop", {}))`

JarbasAl committed:
Update __init__.py

JarbasAl committed:
reply

forslund commented:
The tests also needs updating to not expect a matched intent.

forslund commented on __init__.py:
```diff
@@ -14,30 +14,27 @@
+        self.register_fallback(self.handle_fallback, 50)
+        self.stop_words = get_all_vocab("StopKeyword", self.lang)
+        
+    def handle_fallback(self, message):
```
this will fail for "shut up" for example.

Can we do the following instead:

```python
    def handle_fallback(self, message):
        utterance = message.data.get("utterance", "")
        for stop_word in self.stop_words:
            if stop_word in utterance:
                self.emitter.emit(message.reply("mycroft.stop", {}))
                return True
        return False
```

JarbasAl replied:
done

JarbasAl committed:
fallback

JarbasAl committed:
fix missing method

JarbasAl committed:
missing vocab_path

JarbasAl committed:
missing imports

JarbasAl commented:
was testing against my fork where i have a get_vocab method, added it to the skill

will fix tests now

ndarilek commented:
Thanks for doing this. Looking forward to when it lands, going to make talking about bus stops vastly easier! :)

ndarilek commented:
I don't mean for this to seem pushy, but why isn't this merged? Seems like a no-brainer. I wish this skill didn't trigger on each and every use of the word "stop" in conversation.

Thanks!

forslund committed:
Fix resolve_vocab

the resolve_vocab function used a non-existing method to create a
voc-file path, this replaces that with a simple os.path.join()

forslund committed:
Update test cases to work with this as fallback

forslund commented:
@JarbasAl I pushed a fix to make it work and updated the test cases, could you check that all looks good?

JarbasAl commented:
looks good, but maybe could be changed to use the new is_match skill method to check for the vocs

JarbasAl commented:
sorry i forgot about this and never updated the tests

forslund commented:
is_match is a good idea. Sadly it doesn't do voc's from the skill at the moment only from the resource dir in mycroft-core :/

I'll make an updated version and include it here and send a PR to core as well.

forslund committed:
Use is_match equivalent

mycroft-core has added an is_match() method this uses a similar method
adding support for skill specific vocabulary.
